BrainPortal/app/models/portal_task.rb

```diff
@@ -545,92 +545,196 @@ def params_path_value(paramspath) #:nodoc:
     foundvalue
   end
 
-  # Wrapper class around the ActiveRecord errors() method;
-  # this class answers to the same methods as the errors
+  # Wrapper around the ActiveModel::Errors class.
+  # This class answers to the same methods as the errors
   # object but its 'attributes' are actually paramspaths.
-  # See the Rails classes ActiveRecord::Validations and
-  # ActiveRecord::Errors for more information.
+  #
+  # See the Rails classes ActiveModel::Validations and
+  # ActiveModel::Errors for more information.
+  #
+  # The 'generate_errors' method is not implemented.
+
   class ParamsErrors
-    attr_writer :real_errors
+    include Enumerable
+    attr_writer :real_errors, :base
 
-    def on(paramspath) #:nodoc:
-      @real_errors.on(paramspath.to_la_id)
+    def [](paramspath) #:nodoc:
+      @real_errors[paramspath.to_la_id]
     end
 
-    def [](paramspath) #:nodoc:
-      @real_errors.on(paramspath.to_la_id)
+    def []=(paramspath,*args) #:nodoc:
+      @real_errors.add(paramspath.to_la_id,*args)
     end
 
     def add(paramspath,*args) #:nodoc:
       @real_errors.add(paramspath.to_la_id,*args)
     end
 
+    def added?(paramspath, *args) #:nodoc:
+      @real_errors.added?(paramspath.to_la_id,*args)
+    end
+
     def add_on_blank(paramspaths,*args) #:nodoc:
-      @real_errors.add_on_blank(paramspaths.map(&:to_la_id),*args)
+      Array(paramspaths).each do |paramspath|
+        value = @base.params_path_value(paramspath.to_la_id)
+        add(paramspath, "is blank") if value.blank?
+      end
     end
 
     def add_on_empty(paramspaths,*args) #:nodoc:
-      @real_errors.add_on_empty(paramspaths.map(&:to_la_id),*args)
+      Array(paramspaths).each do |paramspath|
+        value = @base.params_path_value(paramspath.to_la_id)
+        is_empty = value.respond_to?(:empty?) ? value.empty? : false
+        add(paramspath, "is empty") if value.nil? || is_empty
+      end
     end
 
-    def add_to_base(*args) #:nodoc:
-      @real_errors.add_to_base(*args)
+    def as_json(*args) #:nodoc:
+      only_params_errors.as_json(*args)
     end
 
-    def size #:nodoc:
-      @real_errors.size
+    def blank? #:nodoc:
+      only_params_errors.blank?
     end
 
-    def count #:nodoc:
-      @real_errors.size
+    def clear #:nodoc:
+      @real_errors.keys.each { |key|
+        if key =~ (/cbrain_task_params_/i)
+          @real_errors.delete(key)
+        end
+      }
     end
 
-    def length #:nodoc:
-      @real_errors.size
+    def count #:nodoc:
+      only_params_errors.size
     end
 
-    def clear #:nodoc:
-      @real_errors.clear
+    def delete(paramspath) #:nodoc:
+      @real_errors.delete(paramspath.to_la_id.parameterize.underscore.to_sym)
     end
 
+    # NOTE: returns a copy of the real errors hash
     def each(&block) #:nodoc:
-      @real_errors.each(&block)
+      only_params_errors.each(&block)
```

Comment:
The behavior of the `.each()` method is wrong: each yield should be given |att,msg| where msg is not a an array, but a single message, even when multiple error messages are associated with the same atttibute.